osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/processors/UpdateFirmwareRequestMessageProcessor.java

```diff
@@ -37,6 +37,23 @@ protected UpdateFirmwareRequestMessageProcessor() {
     super(MessageType.UPDATE_FIRMWARE);
   }
 
+  @Override
+  protected boolean usesDeviceConnection(final Serializable messageObject) {
+    if (messageObject instanceof final UpdateFirmwareRequestDto requestDto) {
+      final String firmwareIdentification = requestDto.getFirmwareIdentification();
+      final boolean usesDeviceConnection =
```

Comment:
Catching the return value of this method (isFirmwareFileAvailable) in the (usesDeviceConnection) variable hides an implicite business rule. Better make it more explicite.